docs/src/_docs/samplers.jmd

```diff
@@ -0,0 +1,246 @@
+
+```julia
+c = sample(model, NUTS(1000, 0.95))
+plot_sampler(c)
+```
+
+An exceptionally low acceptance rate will show very few moves on the posterior:
+
+```julia
+c = sample(model, NUTS(1000, 0.2))
+plot_sampler(c)
+```
+
+### PG
+
+The Particle Gibbs (PG) sampler is an implementation of an algorithm from the paper "Particle Markov chain Monte Carlo methods" by Andrieu, Doucet, and Holenstein (2010). The interested reader can learn more [here](https://rss.onlinelibrary.wiley.com/doi/full/10.1111/j.1467-9868.2009.00736.x).
+
+The two parameters are the number of particles, and the number of iterations. The plot below shows the use of 20 particles.
+
+```julia
+c = sample(model, PG(20, 1000))
+plot_sampler(c)
+```
+
+Next, we plot using 50 particles.
+
+```julia
+c = sample(model, PG(50, 1000))
+plot_sampler(c)
+```
+
+### PMMH
+
+The Particle Marginal Metropolis-Hastings (PMMH) sampler is an implementation of an algorithm from the paper "Particle Markov chain Monte Carlo methods" by Andrieu, Doucet, and Holenstein (2010). The interested reader can learn more [here](https://rss.onlinelibrary.wiley.com/doi/full/10.1111/j.1467-9868.2009.00736.x).
+
+PMMH supports the use of different samplers across different parameter spaces, similar to the Gibbs sampler. The plot below uses SMC and MH.
+
+```julia
+c = sample(model, PMMH(1000, SMC(20, :m), MH(10,:s)))
+plot_sampler(c)
+```
+
+### PIMH
+
+In addition to PMMH, Turing also support the Particle Independent Metropolis-Hastings (PIMH). PIMH accepts a number of iterations, and an SMC call.
+
+```julia
+c = sample(model, PIMH(1000, SMC(20)))
+plot_sampler(c)
+```
+
+### SGHMC
+
+Stochastic Gradient Hamiltonian Monte Carlo (SGHMC) tends to produce sampling paths not unlike that of stochastic gradient descent in other machine learning model types. It is an implementation of an algorithm in the paper "Stochastic Gradient Hamiltonian Monte Carlo" by Chen, Fox, and Guestrin (2014). The interested reader can learn more [here](https://arxiv.org/abs/1402.4102).
+
+
+The two parameters used in SGHMC are the learing rate and the momentum decay. Here is sampler with a higher momentum decay of 0.1:
+
+```julia
+c = sample(model, SGHMC(1000, 0.001, 0.1))
+plot_sampler(c)
+```
+
+And the same sampler with a much lower momentum decay:
+
+```julia
+c = sample(model, SGHMC(1000, 0.001, 0.01))
+plot_sampler(c)
+```
+
+### SGLD
+
+The Stochastic Gradient Langevin Dynamics (SGLD) is based on the paper "Bayesian learning via stochastic gradient langevin dynamics" by Welling and Teh (2011). A link to the article can be found [here](https://dl.acm.org/citation.cfm?id=3104568).
+
+```julia
+c = sample(model, SGLD(1000, 0.01))
+plot_sampler(c)
+```
+
+### SMC
+
+The Sequential Monte Carlo (SMC) method accepts a number of iterations to use, and tends to be less active than some of the Hamiltonian methods show in above samplers.
+
+```julia
+c = sample(model, SMC(1000))
```

Comment:
SMC is similar to IS, all samples are weighted so we don't usually plot the scattering/path.